### DISCUSSION TITLE
google-earth-pro: program does not start (private-bin)

### DISCUSSION THREAD

**Discussion by firejailaddssecuirty** [netblue30/firejail#4683] [Q&A] [answered]
Just installed google-earth-pro-stable_current_amd64.deb & ran sudo firecfg but Google Earth won't launch.
`$ google-earth-pro 
Reading profile /etc/firejail/google-earth-pro.profile
Reading profile /etc/firejail/google-earth.profile
Reading profile /etc/firejail/disable-common.inc
Reading profile /etc/firejail/disable-devel.inc
Reading profile /etc/firejail/disable-exec.inc
Reading profile /etc/firejail/disable-interpreters.inc
Reading profile /etc/firejail/disable-passwdmgr.inc
Reading profile /etc/firejail/disable-programs.inc
Reading profile /etc/firejail/whitelist-common.inc
Warning: networking feature is disabled in Firejail configuration file
Parent pid 11909, child pid 11910
Private /opt installed in 329.16 ms
8 programs installed in 6.91 ms
Warning: cleaning all supplementary groups
Warning: cleaning all supplementary groups
Warning: cleaning all supplementary groups
Warning: cleaning all supplementary groups
Child process initialized in 415.36 ms
/usr/bin/google-earth-pro: line 21: readlink: command not found
/usr/bin/google-earth-pro: line 21: ./googleearth-bin: No such file or directory

Parent is shutting down, bye...
`
How do I fix this ?

**Comment by rusty-snake** [comment, marked as answer]
> /usr/bin/google-earth-pro: line 21: readlink: command not found

Add it to `private-bin`.

**Reply by firejailaddssecuirty** [reply]
I found another thread about this same issue. I have created a google-earth.local and added the following
`private-bin readlink,dirname,basename,grep,sed`

Should I delete everything other that private-bin ?

**Reply by rusty-snake** [reply]
Does it work now? Based on your reply no. Is there a new error?

**Reply by firejailaddssecuirty** [reply]
Yes, its working. No errors visible. This is the output

```
$ google-earth-pro 
Reading profile /etc/firejail/google-earth-pro.profile
Reading profile /etc/firejail/google-earth.profile
Reading profile /home/home/.config/firejail/google-earth.local
Reading profile /etc/firejail/disable-common.inc
Reading profile /etc/firejail/disable-devel.inc
Reading profile /etc/firejail/disable-exec.inc
Reading profile /etc/firejail/disable-interpreters.inc
Reading profile /etc/firejail/disable-passwdmgr.inc
Reading profile /etc/firejail/disable-programs.inc
Reading profile /etc/firejail/whitelist-common.inc
Warning: networking feature is disabled in Firejail configuration file
Parent pid 1758, child pid 1759
Private /opt installed in 665.63 ms
13 programs installed in 8.86 ms
Warning: cleaning all supplementary groups
Warning: cleaning all supplementary groups
Warning: cleaning all supplementary groups
Warning: cleaning all supplementary groups
Child process initialized in 709.11 ms
```
I want maximum possible isolation/security. So I am curious if these are really necessary ? `readlink,dirname,basename,grep,sed`

---

EDIT by @rusty-snake: code-block formatting

**Reply by rusty-snake** [reply]
> I want maximum possible isolation/security.

Then is firejail the wrong tool for you.

>  So I am curious if these are really necessary ? readlink,dirname,basename,grep,sed

If they are used in the startup script, yes. Open it and analyse it. Maybe you can skip it by manually setting some envvars or so.

What bad do you expect for these binaries?

**Reply by firejailaddssecuirty** [reply]
If an user is looking for maximum isolation which is the proper way of achieving that ?  
I don't trust Google. I almost never install any package outside of my distro's repos. GE is just an exception. I don't want GE to have any access to my personal data.

**Reply by rusty-snake** [reply]
> If an user is looking for maximum isolation which is the proper way of achieving that ?

[Qubes OS](https://www.qubes-os.org/)

**Reply by firejailaddssecuirty** [reply]
I know about Qubes but unfortunately can't use due to low RAM. But there is one big problem with Qubes. Qubes passes all internet traffic via Tor. Suppose I login in to my Gmail or Facebook account using Qubes which is using Tor the whole idea of using Tor is getting defeated. Yes I sometimes use the Tor browser but its not my main browser.

While using Firefox inside firejail I have tested and found that only the Downloads folder is accesible so all other data which in my /home/username is isolated. So firejail does isolate. What's the point that I am missing about firejail ? I mean what is that firejail can't do ?

**Reply by rusty-snake** [reply]
> Qubes passes all internet traffic via Tor.

That's not true.
Whonix does so and Whonix can be used in Qubes OS but you can also use Debian, Fedora, ... in Qubes OS.

> While using Firefox inside firejail I have tested and found that only the Downloads folder is accesible so all other data which in my /home/username is isolated. So firejail does isolate. What's the point that I am missing about firejail ? I mean what is that firejail can't do ?

Firejail add isolation, but if you want "maximum possible isolation", you need a dedicated base-system, kernel and netns (possible with firejail).
